Azure/communication-ui-library-ios
# [Calling][Feature] Implementation of UFD diagnostics notification banners

## Purpose

* This implements handle of User Facing Diagnostics feature in the UISDK as notification banners in call screen following the specified design: https://www.figma.com/file/BOeihN2NJE7ROdyqbihCxi/Mobile-UI-Library?type=design&node-id=7588-30394&mode=design&t=Jp08jcHG0gYzeXF3-0
* 

## Does this introduce a breaking change?



## Pull Request Type
What kind of change does this Pull Request introduce?



## Pull request checklist

This PR has considered:

## How to Test


There a couple of diagnostics a some are hard to test because they indicate a underlying device issue, but a few ones to test may be:
Speaking While Muted
* Join a call
* Mute yourself 
* Start speaking 
* After some time we should see the banner

Network Quality
* Before join a call switch on [network link conditioner iOS feature](https://nshipster.com/network-link-conditioner/) to a bad network.
* Join a call
* Is likely to see a bad network banner 
* Also turn on/off internet should give reconnecting state.

## What to Check

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| Screenshot or description before this change | Screenshot or description with this change |

## Other Information
